### ISSUE TITLE
AttributeError "object has no attribute 'pk'" if "request.user" is not a Django model

### ISSUE THREAD

**Issue by mlegner** [jrobichaud/django-structlog#80]
In my Django application, users are not (always) associated with a model. In this case, I get an `AttributeError` in [`django_structlog/middlewares/request.py#L108`](https://github.com/jrobichaud/django-structlog/blob/master/django_structlog/middlewares/request.py#L108) because the `request.user` does not have an attribute `pk`.

A possible solution would be to use the `pk` attribute if it exists and otherwise just take the `str` of the `user`:

```diff
-            user_id = request.user.pk
+            user_id = getattr(request.user, "pk", str(request.user))
```

**Comment by jrobichaud** [comment]
I'd probably have to do it this way:

```diff
-            user_id = request.user.pk
+            if hasattr(request.user, "pk"):
+                user_id = request.user.pk          
+            else:
+                user_id = str(request.user)
```

There is a subtle difference `getattr` requires `str(request.user)` to be evaluated at every request, and it will serialize the user even though it has a `pk`.

**Comment by jrobichaud** [comment]
I might just have to prevent the crash and let the developers bind what they need using the signals.

**Comment by jrobichaud** [comment]
Not everyone will want to bind the user like you suggested.

**Comment by mlegner** [comment]
> I might just have to prevent the crash and let the developers bind what they need using the signals.

> Not everyone in the same situation as you will want to bind the user like you suggested.

Fully agree. My suggestion was just the simplest fix that I could think of. Just keeping the `user_id` empty and letting the developers assign it would work also.

**Comment by jrobichaud** [comment]
fixed in 3.0.1
See the [doc](https://django-structlog.readthedocs.io/en/latest/events.html#request-bound-metadata) for the signals: to bind your custom `user_id`.
